cpp/pipelines/add_authority_beacon_information.cc

```diff
@@ -21,15 +21,65 @@
 #include <cstdlib>
 #include "BeaconFile.h"
 #include "MARC.h"
+#include "FileUtil.h"
 #include "StringUtil.h"
 #include "util.h"
+#include <iostream>
 
 
 namespace {
 
 
+class TypeFile {
+
+    struct Entry {
+        std::string gnd_number_;
+        std::vector<std::string> types_;
+    public:
+        Entry() = default;
+        Entry(const Entry &other) = default;
+        Entry(const std::string gnd_number, const std::vector<std::string> &types) : gnd_number_(gnd_number), types_(types) {}
+        inline bool operator<(const Entry &rhs) { return gnd_number_ < rhs.gnd_number_; }
+        inline bool operator==(const Entry &rhs) const { return gnd_number_ == rhs.gnd_number_; }
+        inline bool operator()(const Entry &rhs) { return gnd_number_ < rhs.gnd_number_; }
+    };
+
+    class EntryHasher {
+    public:
+        inline std::size_t operator()(const Entry &entry) const { return std::hash<std::string>()(entry.gnd_number_); }
+    };
+
+    private:
+        const std::string filename_;
+        std::unordered_set<Entry, EntryHasher> entries_;
+    public:
+        typedef std::unordered_set<Entry, EntryHasher>::const_iterator const_iterator;
+    public:
+        explicit TypeFile(const std::string filename) : filename_(filename) {
+            unsigned line_no(0);
+            const auto input(FileUtil::OpenInputFileOrDie(filename));
+            while (not input->eof()) {
+                const std::string line(input->getLineAny());
+                ++line_no;
+                std::vector<std::string> gnd_and_types;
+                StringUtil::Split(line, std::string(" - "), &gnd_and_types);
+                if (gnd_and_types.size() != 2)
```

Comment:
You can use the return value of Split itself here.

Reply:
Done